### PR TITLE
Upgrade Cypress

### PR DESCRIPTION
**Pull request by nh758**
Upgrade to the latest Cypress `12.5.1`.
This is up 4 major version.
## Changes to match new Version:
- New cypress config format
- `integrations` folder renamed `e2e`
- Test files should now be named with `.cy.js` 
## Also refactored 
- Our `command.js` helper are now Cypress Commands: Can now use them as `cy.AuthLogin()`, `cy.ResetDB()` and `cy.RunSQL(folder, files)`
- AuthLogin now uses `cy.Session()`
- RunSQL uses `cy` commands, so its easier to debug.
- Most install specific values (user, db password, url, tenant, stack, etc.) are now use `Cypress.env`. These are read from `.env` so we don't need to patch 5 different files.

## See also
https://github.com/digi-serve/ab-cli/pull/48

### CODE REVIEW

File: test/setup/ci-test.overide.yml [hunk 1]
```diff
@@ -29,7 +33,8 @@ services:
       - AB_TESTING=true
```
Comment (Hiro-Nakamura) [inline test/setup/ci-test.overide.yml:33]: Now that we have a `NODE_ENV=true` added as well, can we refactor `AB_TESTING` to use that instead?

Comment (Hiro-Nakamura) [reply test/setup/ci-test.overide.yml:33]: Or is there a value in having 2 different testing flags?

**Review by Hiro-Nakamura — CHANGES_REQUESTED** [review]
If you think everything is OK as is, then good to go.

File: package.json [hunk 2]
```diff
@@ -18,15 +19,15 @@
     "boot": "docker stack deploy -c docker-compose.yml ab & ./node_modules/.bin/forever start app.js",
     "dev-compose": "docker-compose -f docker-compose.dev.yml up & node_modules/.bin/nodemon app.js",
     "dev": "node_modules/.bin/nodemon app.js & docker stack deploy -c docker-compose.dev.yml ab",
-    "cypress:open": "cypress open --project ./test/e2e",
+    "cypress:open": "env-cmd cypress open --project ./test/e2e",
     "test": "appbuilder test setup ab && npm run test:wait &&  npm run test:e2e",
-    "test:boot": "docker stack deploy -c docker-compose.dev.yml -c ./test/setup/test-compose.yml test_ab",
+    "test:boot": "env-cmd docker stack deploy -c docker-compose.dev.yml -c ./test/setup/test-compose.yml test_130",
```
[the remark this answers is not in the document]
Comment (nh758) [reply package.json:24]: no, I missed this, it should still be ab and get's patched by the installer.

File: test/setup/ci-test.overide.yml (hunk 1 above)
Comment (nh758) [reply test/setup/ci-test.overide.yml:33]: I don't think we need `NODE_ENV=test`. I think it was from something else I was trying.

**Review by guyyoo — APPROVED** [review]
LGTM

**Pull request merged**